### PR TITLE
Pal 949 sq develop smells

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: filtered-resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/filteredresource/web/AkkaWebsocketTest.java [hunk 1]
```diff
@@ -50,15 +50,15 @@
...
 import uk.gov.gchq.palisade.service.filteredresource.repository.TokenOffsetController;
 import uk.gov.gchq.palisade.service.filteredresource.repository.TokenOffsetController.TokenOffsetCommand;
 import uk.gov.gchq.palisade.service.filteredresource.repository.TokenOffsetPersistenceLayer;
-import uk.gov.gchq.palisade.service.filteredresource.service.WebsocketEventService;
+import uk.gov.gchq.palisade.service.filteredresource.service.WebSocketEventService;
 import uk.gov.gchq.palisade.service.filteredresource.stream.config.AkkaRunnableGraph.AuditServiceSinkFactory;
 import uk.gov.gchq.palisade.service.filteredresource.stream.config.AkkaRunnableGraph.FilteredResourceSourceFactory;
 import uk.gov.gchq.palisade.service.filteredresource.web.AkkaHttpServer;
-import uk.gov.gchq.palisade.service.filteredresource.web.router.WebsocketRouter;
+import uk.gov.gchq.palisade.service.filteredresource.web.router.WebSocketRouter;
```
Comment (m78233) [inline filtered-resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/filteredresource/web/AkkaWebsocketTest.java:61]: Could not get this to link to line 81, the class name so I put it here.  Suggest you use the camel case with the test classes as well.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/model/WebSocketMessage.java [hunk 2]
```diff
@@ -33,17 +34,17 @@
 import java.util.StringJoiner;
 
 /**
- * WebsocketMessage contains the content of the message sent to the websocket and is used by {@link uk.gov.gchq.palisade.service.filteredresource.service.WebsocketEventService}
+ * WebSocketMessage contains the content of the message sent to the webSocket and is used by {@link WebSocketEventService}
  */
-public final class WebsocketMessage {
+public final class WebSocketMessage {
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/model/WebSocketMessage.java:39]: A few of the Java doc in this class could  be updated with the use of camel case when referring  to WebSocketMessage.  Also, would have expected a test class for this.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/service/WebSocketEventService.java [hunk 3]
```diff
@@ -50,8 +50,8 @@
  * <!--  <li> send any "late" errors to the client (eg. resource-service or policy-service exceptions) -->
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/service/WebSocketEventService.java:50]: Looks like text that was meant to be commented out.  If so remove it.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/model/AuditableWebSocketMessage.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Optional;
+
+/**
+ * This class is a container for a {@code WebSocketMessage} and a {@code Pair} of {@code FilteredResourceRequest}
+ * and {@code CommittableOffset} during stream processing.
+ * Under normal conditions only one of these will be non-null, indicating failed or successful processing.
+ */
+public class AuditableWebSocketMessage {
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/model/AuditableWebSocketMessage.java:33]: Would have expected a corresponding test class for this.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/service/WebSocketEventService.java [hunk 5]
```diff
@@ -50,8 +50,8 @@
...
  * </ul>
  */
-public class WebsocketEventService {
-    private static final Logger LOGGER = LoggerFactory.getLogger(WebsocketEventService.class);
+public class WebSocketEventService {
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/service/WebSocketEventService.java:53]: Would have expected a set of tests for this.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/AkkaHttpServer.java [hunk 6]
```diff
@@ -31,6 +31,9 @@
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+/**
+ * Class to create an http server for akka
+ */
 public class AkkaHttpServer {
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/AkkaHttpServer.java:37]: I think that the Javas Doc in this class needs a bit more to describe the class and methods.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/router/KafkaRestWriterRouter.java [hunk 7]
```diff
@@ -36,9 +36,17 @@
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
+/**
+ * Class to deal with rest messages
+ */
 public class KafkaRestWriterRouter implements RouteSupplier {
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/router/KafkaRestWriterRouter.java:42]: I think that the there could be more added to the Java Doc.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/router/RouteSupplier.java [hunk 8]
```diff
@@ -20,6 +20,9 @@
 
 import java.util.function.Supplier;
 
+/**
+ * Interface for supplying akka routes
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/router/RouteSupplier.java:24]: Need more to explain a RouteSupplier and then expand on this in each of the implementations.

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/router/SpringHealthRouter.java [hunk 9]
```diff
@@ -31,6 +31,9 @@
 import java.util.Map;
 import java.util.Optional;
 
+/**
+ * Class to provide the health of the Spring application
+ */
 public class SpringHealthRouter implements RouteSupplier {
```
Comment (m78233) [inline filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/web/router/SpringHealthRouter.java:37]: I think that this class should have a test and it needs more content in the documentation.

**Review by m78233 — CHANGES_REQUESTED** [review]
The changes do as promised on the tin and the code smells are gone.  There are few places where the documentation could be improved to better explain the classes and the methods.  There were a number of classes that I think should have had corresponding tests, but that was not within the remit of this ticket so after a few, I stopped commenting on this.

File: filtered-resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/filteredresource/web/AkkaWebsocketTest.java (hunk 1 above)
Comment (developer01189998819991197253) [reply filtered-resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/filteredresource/web/AkkaWebsocketTest.java:61]: Updated Websocket within this file where appropriate to WebSocket

File: filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/model/WebSocketMessage.java (hunk 2 above)
Comment (developer01189998819991197253) [reply filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/model/WebSocketMessage.java:39]: Updated Websocket in this class to WebSocket

**Comment by developer01189998819991197253** [comment]
This PR should only be fixing Code smells that are currently breaking develop, the refinement of javadoc or addition of test classes is not required now, and if actually required then a ticket should be created on the backlog for the implementation/fix

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by m78233 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**